UI/MapUI.gd

```diff
@@ -55,12 +58,8 @@ func _add_torch_to_current_location() -> void:
 		for room_light_index: int in range(minimum_room_light_index, maximum_room_light_index):
 			var room_ui: RoomUI = room_ui_array[floor_light_index][room_light_index]
 			if room_ui != null:
-				if floor_light_index == current_player_floor_index + 1 and room_ui.room.light_level == Enums.LightLevel.DIMLY_LIT:
-					room_ui.room.light_level = Enums.LightLevel.BRIGHTLY_LIT
-				else:
-					room_ui.room.light_level = Enums.LightLevel.DIMLY_LIT
+				room_ui.room.increase_light_level()
```

Comment:
I realized we can't really determine light level within RoomBase since it depends if there's a torch in range or now, so I ended up scrapping it and just set the light level directly. If there's a better way to set it, I'm all ears.

Reply:
I don't really see the problem
When you put a torch in a room, take the list of the rooms in the range of the torch (which you get via map movement).
Look at the light level and do the following:
- 0 -> 2
- 1 -> 2
- 2 -> 3
- 3 -> 3

Also when the player move, you also generate the list of new_accessible rooms and you do that on the light level:
- 0 -> 1
- 1 -> 1
- 2 -> 2
- 3 -> 3

Basically you just add one to the light level if it's unlit

i'm not sure to understand why that's not possible to do at the `RoomBase` level